src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int CAPACITY_INCREASE = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+        if (size >= threshold) {
+            resize();
+        }
+        int index;
+        if (key == null) {
+            index = 0;
+        } else {
+            index = hashCode(key) % table.length;
+        }
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = new Node<>(hashCode(key), key, value, null);
```

Comment:
It is not necessary to calculate the hash code again here since it's already calculated in line 26. You can store the hash in a local variable and reuse it.